build_tools/cmake/build_android.sh

```diff
@@ -29,22 +29,146 @@ else
   mkdir "${BUILD_ANDROID_DIR}"
 fi
 
-declare -a args=(
+declare -a common_args=(
   -G Ninja
   -B "${BUILD_ANDROID_DIR}"
   -DCMAKE_TOOLCHAIN_FILE="${ANDROID_NDK}/build/cmake/android.toolchain.cmake"
   -DANDROID_ABI="${ANDROID_ABI}"
   -DANDROID_PLATFORM=android-29
   -DIREE_HOST_BINARY_ROOT="${IREE_HOST_BINARY_ROOT}"
-  -DIREE_ENABLE_ASSERTIONS=ON
   -DIREE_BUILD_COMPILER=OFF
-  -DIREE_BUILD_TESTS=ON
   -DIREE_BUILD_SAMPLES=OFF
 )
 
-# Configure towards 64-bit Android 10, then build.
-"${CMAKE_BIN}" "${args[@]}"
+# Settings for the minimal build used for code size / c++ symbols checks.
+# See "### C++ symbols checks" section below.
+declare -a minimal_build_args=(
+  -DIREE_BUILD_TESTS=OFF
+  # A bunch of instrumentation is tied to IREE_ENABLE_ASSERTIONS - this isn't
+  # just about eliding assertions.
+  -DIREE_ENABLE_ASSERTIONS=OFF
+  # The Vulkan driver is C++.
+  -DIREE_HAL_DRIVER_VULKAN=OFF
+)
+
+# These should at least undo any non-default settings in minimal_build_args.
+declare -a full_build_args=(
+  -DIREE_BUILD_TESTS=ON
+  -DIREE_ENABLE_ASSERTIONS=ON
+  -DIREE_HAL_DRIVER_VULKAN=ON
+)
+
+#############################################################################
+### C++ symbols checks
+#############################################################################
+
+# Checks that a minimal C program exercising the IREE runtime doesn't contain
+# any C++ symbols.
```

Comment:
This would fit in better as a lint check and/or a new script (we also already have https://github.com/iree-org/iree/blob/main/build_tools/cmake/build_runtime_small.sh), not as part of `build_android.sh`.